joomla/joomla-cms
# [5.0] [Guided Tours] Add context to core tours

Pull Request for Issues #40023 and #41273.

### Summary of Changes

Addition of context to the core tours.

### Testing Instructions

Check the list of tours on different pages of the administration. 
When going to 'Articles', the list of tours should present the article and the categories tours in priority.

Special case with categories:
Articles and Categories tours are now showing in each other's context.

Create a duplicate of the categories tour.
Change the title to 'Contact category tour'.
Change the url to `administrator/index.php?option=com_categories&view=categories&extension=com_contact`.
In component selector, select 'categories' and 'contacts'.
Publish the tour and save.

Go to the contacts tour.
Add the 'categories' to the component selector. Save.

You should now have the new category tour available to the contacts and contact categories as tours that show in priority in the list of tours. 
Check the articles and article categories page to make sure the new category tour for contacts is NOT present.
Check the feeds category page to make sure no tour has priority there.

### Actual result BEFORE applying this Pull Request

The numbers of tours showing in the dropdown can go up to X items, X being set in the tours administrator module.
Only the guided tours have context and show first in the list of tours when the user is listing tours or steps.
The tours follow the order in which they have been ordered in the Guided Tours component.

### Expected result AFTER applying this Pull Request

On any given page where the module 'Take a tour' is showing, the tours being in the context of the page are the ones showing in priority.
The tours are still following the order that is set in the Guided Tours component. The only difference is that now there is a pre-list of tours, more easily in reach when a tour exists for the context the user is in.
A tour having the parameter 'extensions' set to 'all' will always show in the top list.
Then are shown the next X tours, X being specified in the administrator module for the tours. Here X is set to 5. The contextual  tours are not part of the count.

![image](https://github.com/joomla/joomla-cms/assets/5964177/c034810b-8bea-4976-b346-a9acc17e65bc)


If X is set to 0, only the contextual tours will show, which gives administrators the possibility to only show contextual tours in the list of tours.

![image](https://github.com/joomla/joomla-cms/assets/5964177/b342035b-4a1c-411a-b14c-688efb7e624d)


### Link to documentations
Please select:
[Admin Module: Guided Tours](https://docs.joomla.org/Help5.x:Admin_Modules:_Guided_Tours)
[Guided Tours: New or Edit Tour](https://docs.joomla.org/Help5.x:Guided_Tours:_New_or_Edit_Tour)
